includes/class-widget.php

```diff
@@ -38,6 +38,22 @@ public function widget( $args, $instance ) {
 		}
 
 		global $post;
+
+		$hide_republication_widget_on_post = get_post_meta( $post->ID, 'republication-tracker-tool-hide-widget' );
+
+		if( $hide_republication_widget_on_post ){
+			$hide_republication_widget_on_post = $hide_republication_widget_on_post[0];
+		}
+
+		$hide_republication_widget_filter = apply_filters( 'hide_republication_widget', false );
```

Comment:
Why apply the filter to default-false when we could instead apply the filter to the value from the post meta? And should the filter pass the `global $post` as a second parameter on the filter, like in these examples?

https://github.com/INN/largo/blob/512da701664b329f2f92244bbe54880a6e146431/inc/ajax-functions.php#L203-L221

https://github.com/INN/super-cool-ad-inserter-plugin/blob/1da4ea864918c590a4bdb0afc7e1e0178c632b27/inc/scaip-shortcode-inserter.php#L130-L144

Reply:
> Why apply the filter to default-false when we could instead apply the filter to the value from the post meta?

That's a good point I didn't think of. I'll update it.

Reply:
> And should the filter pass the global $post as a second parameter on the filter, like in these examples?

@benlk What reason do you think we'd need `global $post` for in the filter?

Reply:
Updated to use the post meta as the default filter value in https://github.com/INN/republication-tracker-tool/pull/54/commits/c7889fbabe274437c517b292b18bf2bfebee5917.

Reply:
We should pass the current $post because filtering functions should have the post explicitly passed to them that they're operating on. I've been bitten too many times by situations where we expected global $post to be the correct post for the context that a function was running in, only to have global $post somehow not be correct. Not passing posts explicitly is a bad pattern.

Reply:
@benlk https://github.com/INN/republication-tracker-tool/pull/54/commits/fa3d283fa778bed9abf556a252558bb187f53b07 now passes the current `$post` into the filter.

Reply:
👍